source/development/testing.rst

```diff
@@ -1,6 +1,21 @@
 Testing
 =======
-
+The primary goal of the Bao Test Framework is to optimize the testing process
+of Bao. It encompasses three core components: (i) a C code library designed for
+real-time test handling, (ii) a build system based on Nix, and (iii) a Python
+API for comprehensive test management.
+
+1. **C Code Library** - This library comprises a set of utilities that simplify
+   test definition and facilitate result assessment, including streamlined
+   result logging;
+
+2. **Nix Build System** - The Nix Build System introduces an abstraction layer
+   for system builds, enabling the construction of various components, from the
+   firmware to the guests running in each test;
+
+3. **Python API** - The Python API adds a layer of abstraction to test
```

Comment:
Do we really want to call this an API? Isn't this actually a tool that is invoked through the command line?

Reply:
You're correct, updated to "Python Tool".